### PR TITLE
Adding static cache in _hdbt_set_site_information

### PR DESCRIPTION
**Pull request by oviag**
Issue: https://helsinkisolutionoffice.atlassian.net/browse/PLAENH-940

Added cacheing for config as we observed from profiling that the function is called numerous times.

### CODE REVIEW

**Review by dragos-dumi-ibm — APPROVED** [review]
looks fine

File: hdbt.theme [hunk 1]
```diff
@@ -1021,16 +1021,23 @@ function hdbt_preprocess_paragraph(array &$variables): void {
  *   Variables array.
  */
 function _hdbt_set_site_information(array &$variables) {
-  $language_id = \Drupal::languageManager()->getCurrentLanguage(LanguageInterface::TYPE_CONTENT)->getId();
+  // Store in static cache to save computation time.
+  $site_info = &drupal_static(__FUNCTION__);
```
Comment (tuutti) [inline hdbt.theme:1025]: Most of these are statically cached already, like config storage and language negotiation, but if we want to do it anyways then we probably should cache everything, like:

```php
function _hdbt_set_site_information(array &$variables) : void {
  static $values = [];

  if (!$values) {
    $language_id = \Drupal::languageManager()->getCurrentLanguage(LanguageInterface::TYPE_CONTENT)->getId();

    // Set site_name as variable.
    $site_config = \Drupal::config('system.site');
    $values['site_name'] = $site_config->get('name');

    $values['city_name'] = t('City of Helsinki');

    // Set site front page absolute url as variable.
    $values['site_front_url'] = Url::fromRoute('<front>', [], ['absolute' => TRUE]);

    /** @var \Drupal\helfi_api_base\Environment\EnvironmentResolver $resolver */
    $resolver = \Drupal::service('helfi_api_base.environment_resolver');

    // Return if the current project is not in environments list.
    try {
      $activeEnvironment = $resolver->getActiveEnvironment()->getEnvironmentName();
      $environment = $resolver->getEnvironment(Project::ETUSIVU, $activeEnvironment);
    }
    catch (\Exception $e) {
      // Set Etusivu absolute url as variable.
      $values['frontpage_instance_url'] = Url::fromUri("https://www.hel.fi/$language_id/");
      return;
    }

    // Revert to English path, if current language path is not found.
    try {
      $url = $environment->getUrl($language_id);
    }
    catch (\Exception $e) {
      $url = $environment->getUrl('en');
    }

    // Set Etusivu absolute url as variable.
    $values['frontpage_instance_url'] = Url::fromUri($url);
  }
  $variables += $values;
}
```

**Review by tuutti — CHANGES_REQUESTED** [review]

**Review by tuutti — APPROVED** [review]
I didn't actually test this, but looks good otherwise

**Comment by oviag** [comment]
> I didn't actually test this, but looks good otherwise

I did test it, and didn't find any flaws, works the same.

**Pull request merged**